src/prng/chacha.rs

```diff
@@ -240,27 +226,22 @@ impl CryptoRng for ChaChaRng {}
 
 impl SeedFromRng for ChaChaRng {
     fn from_rng<R: Rng>(mut other: R) -> Result<Self, Error> {
-        let mut key = [0; KEY_WORDS];
-        for word in key.iter_mut() {
-            *word = other.next_u32();
+        let mut seed = [0u32; SEED_WORDS];
+        unsafe {
+            let ptr = seed.as_mut_ptr() as *mut u8;
+            let slice = slice::from_raw_parts_mut(ptr, SEED_WORDS * 4);
+            other.try_fill(slice)?;
```

Comment:
You're converting from bytes to `u32` here. I think this will produce different results on BE CPUs? We went to all that trouble to make sure `fill_bytes` is portable, so I think this should be too. Maybe the best option is to use `from_seed` here :/